spec/controllers/hq/sponsorships_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Hq::SponsorshipsController, type: :controller do
+  let(:sponsorship) {create :sponsorship, active: true}
+
+  before :each do
+    sign_in instance_double(AdminUser)
+  end
+
+  specify "#inactivate" do
+    expect(Sponsorship).to receive(:find).and_return(sponsorship)
+    put :inactivate, id: sponsorship.id, sponsorship: {end_date: Date.current}
+
+    expect(sponsorship.active).to be false
```

Comment:
Also note that without stubbing it is impossible to test the unhappy path - something that is still missing from this spec.